biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifParser.java

```diff
@@ -1021,7 +1021,8 @@ private Object buildObject(String className, List<String> loopFields, List<Strin
 
 			try {
 				if ( pType[0].getName().equals(Integer.class.getName())) {
-					if ( val != null && ! val.equals("?") && !val.equals(".")) {
+// 					if ( val != null && ! val.equals("?") && !val.equals(".")) {
```

Comment:
Please remove the comment.

Reply:
We acknowledge that the current module doesn’t handle negative numbers and we would soon provide the modified implementation for the same. 

The example scenarios that leads to parsing errors are below:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.827 s <<< FAILURE! - in org.biojava.nbio.structure.io.TestMMCIFWriting
[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.826 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "?"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.843 s <<< FAILURE! - in org.biojava.nbio.structure.io.TestMMCIFWriting
[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.838 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "A"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.816 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "?"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.825 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "N"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.711 s <<< FAILURE! - in org.biojava.nbio.structure.io.TestMMCIFWriting
[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.703 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "?"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.727 s <<< FAILURE! - in org.biojava.nbio.structure.io.TestMMCIFWriting
[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.723 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "THR"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.655 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "50.00"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.814 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "12.307"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] test1A2C(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.723 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "1.00"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1A2C(TestMMCIFWriting.java:77)

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.177 s <<< FAILURE! - in org.biojava.nbio.structure.io.TestMMCIFWriting
[ERROR] test1SMT(org.biojava.nbio.structure.io.TestMMCIFWriting)  Time elapsed: 2.174 s  <<< ERROR!
java.lang.NumberFormatException: For input string: "-6.786"
	at org.biojava.nbio.structure.io.TestMMCIFWriting.testRoundTrip(TestMMCIFWriting.java:153)
	at org.biojava.nbio.structure.io.TestMMCIFWriting.test1SMT(TestMMCIFWriting.java:60) 
```

The logs are retrieved from running the Nondex tool on biojava-structure module.